File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.java
class Solution {
    public int characterReplacement(String s, int k) {
        int res = 0;

        for(int i = 0; i < s.length(); i++) {
            Map<Character,Integer> map = new HashMap<>();
            int maxFreq = 0;
            for(int j = i; j < s.length(); j++) {
                map.put(s.charAt(j), map.getOrDefault(s.charAt(j),0) + 1);
                maxFreq = Math.max(maxFreq, map.get(s.charAt(j)));

                if((j - i + 1) - maxFreq <= k) {
                    res = Math.max(res, j - i + 1);
                }

            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) < 1:
            return 0

        if len(s) == 1:
            return 1

        res = 0
        set_str = set()

        l = 0
        for i in range(len(s)):
            while s[i] in set_str:
                set_str.remove(s[l])
                l += 1

            set_str.add(s[i])

            res = max(res, i - l + 1)

        return res       

                    
        
        